### ISSUE TITLE
Support more @ rules

### ISSUE THREAD

**Issue by WorldSEnder** [futursolo/stylist-rs#29]
Supporting more at rules mainly divides into two (new) categories:

Global at rules that introduce a new name:
- [ ] [`@keyframes`](https://developer.mozilla.org/en-US/docs/Web/CSS/@keyframes): `@keyframes <identifier>` should work similar to the way a new class name is generated, but generate the name used to refer to the animation.
- [ ] [`@counter-style`](https://developer.mozilla.org/en-US/docs/Web/CSS/@counter-style): `@counter-style <identifier>` similar as above, but generate the name referring to the counter.

At rules that are always global and not associated to a name, offering conditional styling:
- [ ] [`@font-face`](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face): Mount font styles. Pretty self explanatory.
- [ ] [`@page`](https://developer.mozilla.org/en-US/docs/Web/CSS/@page): Not supported in all browsers, but sometimes useful - I used it once in the past for a report sheet that could be printed. Can only modify a subset of css attributes of the *document*.
- [ ] [`@font-feature-values`](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-feature-values): Pretty fringe feature. I've never seen it used, but def belongs to this category.

The parser and the ast should be mainly reusable, but emitting the style probably needs to be changed to allow for replacing fragments in at rules. Out of the above `keyframes` and `font-face` would be my two priorities.

**Comment by simbleau** [comment]
Add `@-webkit-keyframes` to the mix.

**Comment by futursolo** [comment]
<img width="786" alt="image" src="https://user-images.githubusercontent.com/11693215/183253333-ad9bcaaf-cbf7-491b-8696-6acb7e5ae83d.png">

<img width="497" alt="image" src="https://user-images.githubusercontent.com/11693215/183253360-919c858a-fd6a-4aac-a554-bce9a485b616.png">

I am not sure why we need to support this since all browser versions that support WebAssembly support `@keyframes` without vendor prefixing.

**Comment by simbleau** [comment]
You're right - I learned this yesterday and forgot to remove my comment here. +1
